lib/concurrent/executor/ruby_thread_pool_executor.rb

```diff
@@ -152,7 +149,7 @@ def ns_execute(*args, &task)
         handle_fallback(*args, &task)
       end
 
-      ns_prune_pool if @next_gc_time < Concurrent.monotonic_time
+      ns_prune_pool
```

Comment:
Unfortunately, this means that it will do even more work on every task posting. I'd rather see it removed completely from the hot path (task posting). Would having pruning scheduled externally twice per idle-time be enough? Is there a reason why do you need this to be more aggressive. The external scheduling would get rid of the problem where it does not prune after a spike when no more tasks are posted.

Reply:
I didn't think it would be a problem.

I guess external scheduling would be fine, but doesn't that mean running a separate thread per thread pool to use as timer ? This seems to be quite a bit of additional complexity ...

In fine, I would be personally fine with keeping the line as it was.

Reply:
I've re-established pruning at @next_gc_time intervals as it was.